tests/unit/plugins/test_authority.py

```diff
@@ -0,0 +1,148 @@
+from gordon_janitor_gcp.clients import gcrm
+from gordon_janitor_gcp.plugins import authority
+
+
+def echoing_helper_coro(data):
+    async def _coro():
+        return data
+    return _coro()
+
+
+@pytest.fixture
+def fake_authority(mocker, monkeypatch, authority_config, auth_client):
+    monkeypatch.setattr('gordon_janitor_gcp.plugins.auth.GAuthClient',
+                        mocker.Mock(return_value=auth_client))
+    rrset_channel = asyncio.Queue()
+    fake_authority = authority.GCEAuthority(
+        authority_config, rrset_channel)
+    fake_authority.session = auth_client._session
+    return fake_authority
+
+
+@pytest.fixture
+def get_mock_client(mocker, get_gce_client):
+    def _create_fake(klass, *args, **kwargs):
+        client = get_gce_client(klass, *args, **kwargs)
+        fake_client = mocker.Mock(client)
+        return fake_client
+    return _create_fake
+
+
+@pytest.mark.asyncio
+async def test_builder_creates_proper_authority(mocker, authority_config):
+    gce_client = mocker.MagicMock(name='gce_client')
+    crm_client = mocker.MagicMock(name='crm_client')
+    auth_client = mocker.MagicMock(name='auth_client')
+    authority_mod_patch = 'gordon_janitor_gcp.clients.'
+    mocker.patch(authority_mod_patch + 'gcrm.GCRMClient', crm_client)
+    mocker.patch(authority_mod_patch + 'gce.GCEClient', gce_client)
+    mocker.patch(authority_mod_patch + 'auth.GAuthClient', auth_client)
+
+    rrset_channel = asyncio.Queue()
+    kwargs = {'other': 'kwargs'}
+    builder = authority.GCEAuthorityBuilder(
+        authority_config, rrset_channel, **kwargs)
+    session = builder.session
+    with contextlib.closing(session):
+        gce_authority = builder.build_authority()
+
+    auth_client_calls = [
+        mocker.call(
+            authority_config['keyfile'],
+            scopes=authority_config['scopes'],
+            session=session),
+    ] * 2
+
+    auth_client.assert_has_calls(auth_client_calls, any_order=True)
+    expected_metadata_blacklist = [
+        dict([pair]) for pair in authority_config['metadata_blacklist']
+    ]
+    gce_client.assert_called_once_with(
+        auth_client(),
+        session,
+        blacklisted_tags=authority_config['tag_blacklist'],
+        blacklisted_metadata=expected_metadata_blacklist)
+
+    assert crm_client() is gce_authority.crm_client
+    assert gce_client() is gce_authority.gce_client
+    assert rrset_channel is gce_authority.rrset_channel
+    assert authority_config is gce_authority.config
+
+
+@pytest.mark.asyncio
+async def test_builder_error_on_required_config(authority_config, caplog):
+    caplog.set_level(logging.ERROR)
+    del authority_config['keyfile']
+    rrset_channel = asyncio.Queue()
+    kwargs = {}
+
+    builder = authority.GCEAuthorityBuilder(
+        authority_config, rrset_channel, **kwargs)
+    session = builder.session
+    with pytest.raises(exceptions.GCPConfigError), contextlib.closing(session):
```

Comment:
May be a bit pedantic, but perhaps match at least part of the error message that's raised?

Reply:
I removed the test since the same behavior was being tested in `test_init.py::test_get_authority_config_raises`.